### PR TITLE
[Merged by Bors] - chore: add VSCode branch protection to master

### PR DESCRIPTION
**Pull request by BoltonBailey**
Adds branch protection to VSCode for the master branch, so users don't commit accidentally.

Screenshot 2024-02-06 at 10 30 14 PM (2)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -10,5 +10,8 @@
   "githubPullRequests.ignoredPullRequestBranches": [
     "master"
   ],
-  "git.ignoreLimitWarning": true
+  "git.ignoreLimitWarning": true,
+  "git.branchProtection": [
+    "master"
```
Comment (Vierkantor) [inline .vscode/settings.json:15]: The following branches are protected according to the GitHub settings:
```suggestion
    "master",
    "stable",
    "nightly-testing-*",
    "bump/v4*"
```
(does `*` work in VS Code as well to match everything?)

Comment (BoltonBailey) [reply .vscode/settings.json]: Let me check

Comment (BoltonBailey) [reply .vscode/settings.json]: Unfortunately, unlike the .gitignore style globs for directory names, it seems * does not work. I will remove those but keep master and stable, I don't necessarily think it's worth remembering to update the VSCode settings every time a nightly-testing- comes out

**Review by Vierkantor — APPROVED** [review]
LGTM, thanks! Is there anything else we should check before moving this out of draft state?

**Comment by BoltonBailey** [comment]
Just needed the conflicting change to be merged so I could resolve the conflict.

**Comment by BoltonBailey** [comment]
Perhaps `staging` should be added as well?

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**